tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-query-construct-type-negative.bal

```diff
@@ -284,4 +284,41 @@ function testConstructingInvalidReadonlyMap() {
 
     map<string> & readonly|error mp4 = map from var item in [["1", 1], ["2", 2], ["3", 3], ["4", 4]]
                                         select item on conflict error("Error");
+
+    [string:Char, int[]][] list = [];
+    map<float[]>|error mp5 = map from var item in list select item;
+
+    map<int[]> & readonly|error mp7 = map from var item in list select item;
+}
+
+type FooBar1 ("foo"|"bar"|int)[2];
+type FooBar2 ("foo"|"bar"|1)[2];
+type FooBar3 "foo"|"bar"|1;
+type FooBar4 "foo"|"bar"|float;
+type FooBar5 "foo"|"bar"|1;
+
+function testInvalidTypeInSelectWithQueryConstructingMap5() {
```

Comment:
Better to have this reflect that it's testing types that are not a subtype of string as the key.